lib/form_personal_details.dart

```diff
@@ -0,0 +1,200 @@
+//!! This might be deprecated.
+// Delete if still not in use after a while.
+
+import 'package:flutter/material.dart';
+import 'package:tutorial/constants.dart';
+import 'package:tutorial/form_box.dart';
+import 'package:tutorial/homepage.dart';
+// import 'package:firebase_ui_auth/firebase_ui_auth.dart';
+
+class PersonalForm extends StatefulWidget {
+  const PersonalForm({super.key});
+
+  @override
+  State<PersonalForm> createState() => _PersonalFormState();
+}
+
+class _PersonalFormState extends State<PersonalForm> {
+  @override
+  void initState() {
+    super.initState();
+
+    _firstNameController = TextEditingController();
+    _lastNameController = TextEditingController();
+    _dateOfBirthController = TextEditingController();
+    _mobileNumberController = TextEditingController();
+
+  }
+
+  @override
+  void dispose() {
+    _firstNameController.dispose();
+    _lastNameController.dispose();
+    _dateOfBirthController.dispose();
+    _mobileNumberController.dispose();
+
+    super.dispose();
+  }
+
+  late TextEditingController _firstNameController;
+  late TextEditingController _lastNameController;
+  late TextEditingController _dateOfBirthController;
+  late TextEditingController _mobileNumberController;
+
+  String _firstNameError = '';
+  String _lastNameError = '';
+  String _dateOfBirthError = '';
+  String _mobileNumberError = '';
+  String required = 'This field is required';
```

Comment:
this is a keyword, I would probably avoid it.  Also, maybe name it thisFieldIsRequired or requiredErrorMessage. Also, can this be private.